index.ts

```diff
@@ -1,31 +1,38 @@
 import { multiUrlGenerator } from './lib/multiUrlGenerator';
 import fetch from 'node-fetch';
+import console from 'console';
+import yargs from 'yargs';
+import { generalTransformer } from './config/parsers/ulr.parser';
+const PATH = './config/url_components/';
+const MODE = yargs.argv._[0];
 
-// Test fetching Wiki articles
-const requestUrls = multiUrlGenerator(
-  'https://$lang.wikipedia.org/wiki/$article',
-  [
-    { $lang: 'en', $article: 'Plato' },
-    { $lang: 'es', $article: 'Aristóteles' },
-  ]
-);
+async function getUrls(file: string) {
+  const requestUrls = multiUrlGenerator(
+    await import(PATH + file)
+      .then((jsonFile) => jsonFile)
+      .then((data) => data.linkto),
 
-async function main() {
-  for (const url of requestUrls) {
+    await import(PATH + file)
+      .then((jsonFile) => jsonFile)
+      .then((data) => data.components)
+  );
+
+  return new Promise<string[]>((resolve, reject) => {
+    resolve(requestUrls);
+    reject('no files found');
+  });
+}
+
+async function main(mode: string) {
+  for (const url of await getUrls(`${mode}.urls.json`).then((x) => x)) {
     try {
-      const response = await fetch(url);
-      console.log(
-        `Called url: ${url} - with response status:`,
-        response.status
-      );
+      const response = await fetch('https://' + url);
+      generalTransformer(response);
```

Comment:
Sorry if this was not clear in the ticket, but my idea included also dynamically importing the parser (transformer) function.


- First we rename the transformer to `wikipedia.parser.ts` and modify it so that it exports a `{ default: [Function] }` object like this:

```typescript
import { Response } from 'node-fetch';

// export = (....) will make this module to export itself as a "default" prop
// similar to: export const default = function(...)
export = function (response: Response): void {
  console.log(`response url: ${decodeURIComponent(response.url)}`);
  console.log(`response status: ${response.status}`);
  console.log(`response status text: ${response.statusText}`);
};
```

- Then we can create a `getParser` function just like we did for `getUrlComponents` to dynamically import the default module.
```typescript
// async function that expect a string and returns the promise of a function that expects a response as its only argument and returns nothing (only side effects)
async function getParser(
  parserFileName: string
): Promise<(arg0: Response) => void> {
  const parser = await import(`${PARSER_PATH}/${parserFileName}`);
  return parser.default;
}
```

- Then you can call it in the main loop like this:
```typescript
async function main(mode: string) {
  const parser = await getParser(`${mode}.parser.ts`);
  const urls = await getUrls(`${mode}.urls.json`);

  for (const url of urls) {
    try {
      const response = await fetch(url);
      parser(response);
    } catch (err) {
      console.warn(err);
    }
  }
  process.exit(0);
}
```